config/config.go

```diff
@@ -1710,10 +1843,62 @@ func (config *Config) GetVersionLabelName(labels map[string]string) (string, boo
 }
 
 func (c *Config) CertPool() *x509.CertPool {
-	if pool := c.Deployment.certPool; pool != nil {
-		return pool.Clone()
+	if c.Credentials == nil {
+		// Fallback to system cert pool
+		log.Warning("CertPool requested but CredentialManager not initialized. Using system cert pool.")
+		systemPool, err := x509.SystemCertPool()
+		if err != nil {
+			log.Warningf("Unable to load system cert pool: %v", err)
+			return x509.NewCertPool()
+		}
+		return systemPool
 	}
-	return nil
+
+	pool := c.Credentials.GetCertPool()
+	if pool == nil {
+		// Fallback if cert pool not initialized
+		log.Warning("CertPool not initialized in CredentialManager. Using system cert pool.")
+		systemPool, err := x509.SystemCertPool()
+		if err != nil {
+			log.Warningf("Unable to load system cert pool: %v", err)
+			return x509.NewCertPool()
+		}
+		return systemPool
+	}
+
+	return pool
+}
+
+// Close cleans up Config resources such as the credential file watcher.
+// Should be called during application shutdown.
+func (c *Config) Close() {
+	if c.Credentials != nil {
+		c.Credentials.Close()
+		c.Credentials = nil
+	}
+}
+
+// GetCredential reads a credential value, supporting both literal values and file paths.
+// If the value starts with "/", it's treated as a file path and read from disk with caching.
+// Otherwise, the literal value is returned as-is.
+//
+// This method supports automatic credential rotation - when files are updated (e.g., Kubernetes
+// secret rotation), the cache is automatically refreshed without requiring a pod restart.
+//
+// If the CredentialManager is not initialized, falls back to direct file reads without caching.
+func (c *Config) GetCredential(value string) (string, error) {
+	if c.Credentials != nil {
+		return c.Credentials.Get(value)
+	}
+	// Fallback for configs without credential manager (e.g., some tests)
```

Comment:
Doing things especially for tests like this usually means the tests aren't performing setup properly.

Reply:
there are now warnings log messages if this happens. but for simple tests we don't need them to setup cred mgr